src/main/java/org/tron/core/config/args/Args.java

```diff
@@ -539,7 +492,8 @@ private static void bindIp(final com.typesafe.config.Config config) {
         .trim().isEmpty()) {
       if (INSTANCE.nodeDiscoveryBindIp == null) {
         logger.info("Bind address wasn't set, Punching to identify it...");
-        try (Socket s = new Socket("www.baidu.com", 80)) {
+        try {
+          Socket s = new Socket("www.baidu.com", 80);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "Socket" in a "finally" clause. ([squid:S2095](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2095))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=862&issues=AWOmY5GrlfhLAgo0Gw_J&open=AWOmY5GrlfhLAgo0Gw_J)

[ISSUE_KEY]: AWOmY5GrlfhLAgo0Gw_J